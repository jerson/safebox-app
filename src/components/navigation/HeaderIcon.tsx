import React from 'react';
import {StyleProp, StyleSheet, ViewStyle, TextStyle} from 'react-native';
import Colors from '../../modules/constants/Colors';

import ButtonIcon from '../ui/ButtonIcon';

const styles = StyleSheet.create({
  back: {
    fontSize: 28,
    color: Colors.grey5,
  } as TextStyle,
  container: {
    height: 50,
    width: 50,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    opacity: 1,
  } as ViewStyle,
});

export interface HeaderIconProps {
  onPress?: () => void;
  name: string;
  style?: StyleProp<TextStyle>;
}

function HeaderIcon({style, name, onPress}: HeaderIconProps) {
  return (
    <ButtonIcon
      name={name}
      iconStyle={[styles.back, style]}
      style={styles.container}
      onPress={onPress}
    />
  );
}

export default HeaderIcon;
