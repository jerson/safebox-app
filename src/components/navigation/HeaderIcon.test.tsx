import React from 'react';
import * as renderer from 'react-test-renderer';
import HeaderIcon, {HeaderIconProps} from './HeaderIcon';

const props: HeaderIconProps = {
  name: 'trash',
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<HeaderIcon {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<HeaderIcon {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
