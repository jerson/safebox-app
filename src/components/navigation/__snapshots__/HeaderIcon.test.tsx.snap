// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#65748b",
            "fontSize": 28,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‡§
  </Text>
</_class>
`;
